src/bokeh/util/serialization.py

```diff
@@ -339,7 +338,7 @@ def _cast_if_can(array: npt.NDArray[Any], dtype: type[Any]) -> npt.NDArray[Any]:
 
     return array
 
-def transform_series(series: pd.Series[Any] | pd.Index | PandasArray) -> npt.NDArray[Any]:
+def transform_series(series: pd.Series[Any] | pd.Index | pd.api.extensions.ExtensionArray) -> npt.NDArray[Any]:
```

Comment:
I was hoping that there's a common base, so that we don't have to handle each array type separately.